src/unstract/sdk/index.py

```diff
@@ -236,13 +236,17 @@ def index_file(
         full_text = []
         extracted_text = ""
         try:
-            x2text = X2Text(tool=self.tool)
-            x2text_adapter_inst: X2TextAdapter = x2text.get_x2text(
-                adapter_instance_id=x2text_adapter
-            )
-            extracted_text = x2text_adapter_inst.process(
-                input_file_path=file_path, output_file_path=output_file_path
-            )
+            if not output_file_path:
+                with open(file_path, encoding="utf-8") as file:
+                    extracted_text = file.read()
```

Comment:
This may not work as expected if the file is a PDF right? I understand this is probably for the use case of quickly extracting text in the context of summaries but can you update the PR description as to why this change was done?

Reply:
@chandrasekharan-zipstack There is no scenario as of now where the input file is a PDF and output path is not present.

Reply:
My concern here is that, the function's behaviour needs to be explicit based on its arguments. The way we've currently handled - the caller needs to inspect the code to understand its behaviour. We could have ideally added a param `extract_text: bool` that can carry out the desired functionality of skipping a call to LLM whisperer.

By continuing to vaguely differentiate logic based on an unrelated param, we'd only find it increasingly difficult to maintain / add features